crates/noirc_abi/src/lib.rs

```diff
@@ -149,52 +156,73 @@ impl Abi {
     pub fn public_abi(self) -> Abi {
         let parameters: Vec<_> =
             self.parameters.into_iter().filter(|param| param.is_public()).collect();
-        Abi { parameters }
+        let param_witnesses = self
+            .param_witnesses
+            .into_iter()
+            .filter(|(param_name, _)| parameters.iter().any(|param| &param.name == param_name))
+            .collect();
+        Abi { parameters, param_witnesses }
+    }
+
+    /// Encode a set of inputs as described in the ABI into a `WitnessMap`.
+    pub fn encode_to_witness(&self, input_map: &InputMap) -> Result<WitnessMap, AbiError> {
+        // TODO: add handling for missing inputs similarly to how we do in `encode_to_array`.
```

Comment:
Noting TODO

Reply:
Addressed in [dcb07a4](https://github.com/noir-lang/noir/pull/851/commits/dcb07a4261da0055d82318e2e9ec5301dd5769a6)